### PR TITLE
Move show password checkbox under password field

### LINKED ISSUE #6722: Move "show password" checkbox on login form
From #6040:

@anowlcalledjosh:
> This means that the password field now takes two <kbd>Tab</kbd>-presses to reach from the username field rather than one, which breaks at least one popular password manager.
>
> If the accessibility issue is more important than avoiding some manual configuration of password managers, then fine (I'm not qualified to make that decision), but the original change to add `tabindex`es was intentional – see #2824, #2917.

@di:
> This has actually been bugging me too. I think we can fix it without adding tab indexes by moving the "Show password" box under the password field.
> 
> @nlhkabu @woodruffw Thoughts?

@woodruffw:
> Yep, I think this would be a good solution.

---

**Good First Issue**: This issue is good for first time contributors. If you've already contributed to Warehouse, work on [another issue without this label](https://github.com/pypa/warehouse/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+-label%3A%22good+first+issue%22) instead. If there is not a corresponding pull request for this issue, it is up for grabs. For directions for getting set up, see our [Getting Started Guide](https://warehouse.pypa.io/development/getting-started/). If you are working on this issue and have questions, feel free to ask them here, [`#pypa-dev` on Freenode](https://webchat.freenode.net/?channels=%23pypa-dev), or the [pypa-dev mailing list](https://groups.google.com/forum/#!forum/pypa-dev).

### PR DESCRIPTION
**Pull request by sanjaysiddhanti**
@di 

This PR implements https://github.com/pypa/warehouse/issues/6722. 

Screenshot before:

Screen Shot 2020-06-01 at 10 47 17 PM

Screenshot after:

Screen Shot 2020-06-01 at 10 50 33 PM

Let me know if you'd like any additional changes.

Fixes #6722.

### CODE REVIEW

**Comment by sanjaysiddhanti** [comment]
I could also move "show password" down to be horizontally aligned with "Log in" and "forgot password." I wasn't sure where exactly you wanted it. This does make password one tab away from username as the ticket requests.

**Comment by di** [comment]
> I could also move "show password" down to be horizontally aligned with "Log in" and "forgot password."

Yes, please do that! I think that can be done with the `split-layout` class. We should also remove the `split-layout` class where it's no longer being used, like https://github.com/pypa/warehouse/issues/6722#issuecomment-538182534 describes.

**Comment by sanjaysiddhanti** [comment]
Ok, I removed the `split-layout` where it is no longer used. When I apply the rest of the diff from https://github.com/pypa/warehouse/issues/6722#issuecomment-538182534, it basically looks like my screenshot above, without horizontal alignment.

How does d845979 look? I did the following:

- Moved "show password" into the form group that has the log in and forgot password buttons
- This broke the "show password" functionality. Looks like clicking on "show password" triggers some JS code, but it needs to be in a block that connects it to the Password Controller. So, I moved the entire form group into the `<div data-controller="password" class="form-group">` block. Is there a better solution?

New screenshot below

image

File: warehouse/templates/accounts/login.html [hunk 1]
```diff
@@ -86,16 +82,19 @@ <h1 class="page-title">{% trans title=title %}Log in to {{ title }}{% endtrans %
...
+              <div>
+                <input type="submit" value="{% trans %}Log in{% endtrans %}" class="button button--primary">
+              </div>
+              <label for="show-password">
+                <input data-action="change->password#togglePasswords" data-target="password.showPassword"
+                  id="show-password" type="checkbox">&nbsp;{% trans %}Show password{% endtrans %}
+              </label>
+              <span>
+                <a href="{{request.route_url('accounts.request-password-reset')}}">{% trans %}Forgot password?{% endtrans %}</a>
+              </span>
```
Comment (di) [inline warehouse/templates/accounts/login.html:96]: I _think_ wrapping these in a `div` will cause them to both be moved to the right side, but stacked on top of each other

```suggestion
              <div>
                <label for="show-password">
                  <input data-action="change->password#togglePasswords" data-target="password.showPassword"
                    id="show-password" type="checkbox">&nbsp;{% trans %}Show password{% endtrans %}
                </label>
                <span>
                  <a href="{{request.route_url('accounts.request-password-reset')}}">{% trans %}Forgot password?{% endtrans %}</a>
                </span>
              </div>
```

**Comment by sanjaysiddhanti** [comment]
>  will cause them to both be moved to the right side, but stacked on top of each other

Ah, I didn't realize that was the goal. Your suggested diff doesn't quite do it, since `label` and `span` are both inline (see below).

Screen Shot 2020-06-02 at 5 56 23 PM

But when I wrap the `label` in a `div` in b98f393, I think it looks right

Screen Shot 2020-06-02 at 5 58 04 PM

**Comment by sanjaysiddhanti** [comment]
@nlhkabu No rush, but just a friendly ping on this one in case it got lost.

File: warehouse/templates/accounts/login.html [hunk 2]
```diff
@@ -86,12 +82,15 @@ <h1 class="page-title">{% trans title=title %}Log in to {{ title }}{% endtrans %
...
             {% endif %}
           </div>
-        </div>
-
-        <div class="form-group">
-          <div class="split-layout split-layout--table">
-            <div>
-              <input type="submit" value="{% trans %}Log in{% endtrans %}" class="button button--primary">
+          <div class="form-group">
+            <div class="split-layout margin-top--large margin-bottom--large">
```
Comment (sanjaysiddhanti) [inline warehouse/templates/accounts/login.html:86]: removed `split-layout--table`, as it causes elements to be vertically centered in the container, whereas Nicole's screenshot shows "Show password" aligned to the top of the div

**Comment by sanjaysiddhanti** [comment]
Thanks for the feedback! Here is an updated version:

image

The only difference I see is that Nicole's version has double the margin between "Log in" and "Show password." I tried to achieve this with the following, but the margins were getting collapsed with the sibling element:

```
            <span class="display-block margin-top--large">
              <a href="{{request.route_url('accounts.request-password-reset')}}">{% trans %}Forgot password?{% endtrans %}</a>
            </span>
```
I also tried wrapping the `span` in a `div class="margin-top--large"` 

@di did you have an idea of how to get this extra margin and does it matter to you?

**Comment by patelneel55** [comment]
Since there's now a new feature request regarding a "Keep me signed in" button at #8033, would it be beneficial to switch to a :eye: icon, as mentioned by @bittner at [#6722](https://github.com/pypa/warehouse/issues/6722#issuecomment-613121450), to prevent a clutter of text fields under the password field?

Any thoughts? @di @sanjaysiddhanti

**Comment by bittner** [comment]
Nice! Adding an :eye: icon can maybe done easily now. Let's see. :smiley_cat: 

Thanks for this improvement so far! :1st_place_medal:

**Comment by bittner** [comment]
Wow! This change is already rolled out. Nice!

Is the pipeline that does this public? It doesn't seem to be the [GitHub Actions](https://github.com/pypa/warehouse/actions) of this repository. :thinking:

**Comment by brainwane** [comment]
Hi @sanjaysiddhanti! Congrats on landing your first pull request in PyPA projects. Could I ask you to take a look at #5222 and comment there if you can work on it?

**Pull request merged**